app-misc/opentrack/opentrack-9999.ebuild

```diff
@@ -0,0 +1,89 @@
+# Copyright 2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake desktop flag-o-matic toolchain-funcs
+
+DESCRIPTION="Head tracking software for MS Windows, Linux, and Apple OSX"
+HOMEPAGE="https://github.com/opentrack/opentrack"
+
+if [[ ${PV} == 9999 ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/opentrack/opentrack.git"
+else
+	SRC_URI="https://github.com/opentrack/opentrack/archive/refs/tags/${P}.tar.gz"
+	KEYWORDS="~amd64"
+	S="${WORKDIR}/opentrack-${P}"
+fi
+
+LICENSE="ISC"
+SLOT="0"
+
+IUSE="neuralnet wine opencv openmp"
+
+REQUIRED_USE="neuralnet? ( openmp opencv )"
+
+DEPEND="
+	neuralnet? ( sci-libs/onnxruntime-bin )
+	opencv? ( media-libs/opencv:= )
+	wine? ( virtual/wine )
+	dev-libs/libevdev
+	dev-libs/libusb:1
+	dev-qt/qtbase:6
+	sys-process/procps:=
+	x11-libs/libX11
+"
+RDEPEND="${DEPEND}"
+
+pkg_pretend() {
+	[[ ${MERGE_TYPE} != binary ]] && use openmp && tc-check-openmp
+}
+
+pkg_setup() {
+	[[ ${MERGE_TYPE} != binary ]] && use openmp && tc-check-openmp
+}
+
+src_prepare() {
+	# work around hard-coded docs path
+	sed -e 's#share/doc/opentrack#share/doc/'${P}'#g' \
+		-i cmake/*.cmake || die
+
+	cmake_src_prepare
+}
+
+src_configure() {
+	filter-lto
+	use openmp && append-cxxflags -fopenmp
```

Comment:
More stuff is needed for `openmp` support.  There are many examples in the tree, please grep for `tc-check-openmp` and see `BDEPEND`, `pkg_pretend`, `pkg_setup`.

Reply:
Hmm, okay, I will check that. At least, this seems to work. I'll leave this open.

Reply:
I checked multiple ebuilds and none of them set specific BDEPENDs. But I found this:
```
BDEPEND="
	openmp? ( virtual/libomp )
"
```

Is this what you mean? I already added the usual `pkg_pretend` and `pkg_setup` functions.

Reply:
There's another one:
```bash
BDEPEND="
    openmp? (
        || (
            sys-devel/gcc[openmp]
            llvm-core/clang-runtime[openmp]
        )
    )
"
```

I was once told that ebuilds should never depend on gcc or similar packages because this is handled by the system deps. I found this example in oneDNN. Most other package seem to omit that.

So I'm not sure if I could know what is missing here: `pkg_pretend` and `pkg_setup` already replicates what other packages do, and anything I find for `BDEPEND` seems to be an issue of the past.

Reply:
I think you are right, let's skip the BDEPEND for now.  It should help the resolver, but `pkg_pretend` would fail before it nevertheless.